File: LeetCode/2022. Convert 1D Array Into 2D Array/solution.cpp
/**
 * 107 / 107 test cases passed.
 * Runtime: 100 ms
 * Memory Usage: 83.2 MB 
 */
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (original.size() != m * n) return {};
        vector<vector<int>> ret(m, vector<int>(n));
        for (int i = 0; i < m; ++ i) {
            for (int j = 0; j < n; ++ j) {
                ret[i][j] = original[i * n + j];
            }
        }
        return ret;
    }
};
                

File: LeetCode/2022. Convert 1D Array Into 2D Array/solution.py
"""
107 / 107 test cases passed.
Runtime: 88 ms
Memory Usage: 19.8 MB
"""
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if len(original) != m * n:
            return []
        mat = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                mat[i][j] = original[i * n + j]
        return mat
    
